### PR TITLE
Fix CircleCI build and Golang linting

### PR DESCRIPTION
**Pull request by frankmariette**
CI builds are failing due to the certificate change shown [here](https://letsencrypt.org/docs/dst-root-ca-x3-expiration-september-2021/). Updating to the latest image from Circle CI fixes this problem and allows Admiral to build again as expected. 

Fixed linter to run though  did not make changes to any current files to limit this to just fixing CI and not lint issues.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -7,7 +7,7 @@ jobs:
   build:
     working_directory: /go/pkg/mod/github.com/admiral
     docker:
-    - image: circleci/golang:1.11
+    - image: circleci/golang:1.16
```
Comment (nirvanagit) [inline .circleci/config.yml:10]: Should we use 1.17 ?

File: .circleci/config.yml [hunk 2]
```diff
@@ -54,6 +54,7 @@ jobs:
       MINIKUBE_WANTREPORTERRORPROMPT: false
       MINIKUBE_HOME: /home/circleci
       CHANGE_MINIKUBE_NONE_USER: true
+    resource_class: large
```
Comment (nirvanagit) [inline .circleci/config.yml:57]: Curious to know why we need a large build container (assuming it is a build container)

File: tests/create_cluster.sh [hunk 3]
```diff
@@ -5,7 +5,7 @@
 k8s_version=$1
 
 if [[ $IS_LOCAL == "false" ]]; then
-  sudo -E minikube start --vm-driver=none --cpus 4 --memory 4096 --kubernetes-version=$k8s_version &> $HOME/minikube.log 2>&1 < /dev/null
+  sudo -E minikube start --vm-driver=none --cpus 6 --memory 6144 --kubernetes-version=$k8s_version &> $HOME/minikube.log 2>&1 < /dev/null
```
Comment (nirvanagit) [inline tests/create_cluster.sh:8]: Is the increment in resources due to some new dependency?

Comment (frankmariette) [reply tests/create_cluster.sh:8]: Increased demands for the tests with added GRPC clients. I noticed the tests were flaky and seemed to be failing to timeout. Experience with running minikube has shown that the lack of responsiveness occurs when it becomes resource strapped so I increased it to account for this. I also had to increase the integration test vm resources to account for this change as well.

File: .circleci/config.yml (hunk 2 above)
Comment (frankmariette) [reply .circleci/config.yml:57]: This is for the integration tests. The container was increased to match the size that CircleCI has with the limits I increased minikube to. https://circleci.com/docs/2.0/configuration-reference/#resourceclass

**Review by aattuluri — APPROVED** [review]

**Pull request merged**